src/ossia-max/src/fuzzysearch.cpp

```diff
@@ -0,0 +1,99 @@
+// This is an open source non-commercial project. Dear PVS-Studio, please check it.
+// PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+#include <ossia-max/src/fuzzysearch.hpp>
+#include <ossia-max/src/ossia-max.hpp>
+
+using namespace ossia::max;
+
+#pragma mark -
+#pragma mark ossia_fuzzysearch class methods
+
+extern "C" void ossia_fuzzysearch_setup()
+{
+  auto& ossia_library = ossia_max::instance();
+
+  // instantiate the ossia.fuzzysearch class
+  ossia_library.ossia_fuzzysearch_class = class_new(
+      "ossia.fuzzysearch", (method)ossia_fuzzysearch_new, (method)fuzzysearch::free,
+      (long)sizeof(fuzzysearch), 0L, A_GIMME, 0);
+
+  auto& c = ossia_library.ossia_fuzzysearch_class;
+  class_addmethod(
+      c, (method)fuzzysearch::search,
+      "search", A_GIMME, 0);
+  class_addmethod(
+      c, (method)fuzzysearch::assist,
+      "assist", A_CANT, 0);
+  class_addmethod(
+      c, (method)fuzzysearch::notify,
+      "notify", A_CANT, 0);
+
+  class_register(CLASS_BOX, ossia_library.ossia_fuzzysearch_class);
+}
+
+extern "C" void* ossia_fuzzysearch_new(t_symbol* s, long argc, t_atom* argv)
+{
+  auto x = make_ossia<fuzzysearch>(argc, argv);
+  x->m_root = &ossia_max::get_default_device()->get_root_node();
```

Comment:
likely at some point we have to list all devices here instead ?

Reply:
yes sure, this is a quick win :-)
but I'll add an autoconnection mechanism to ossia-max : max will search for new devices every x sec then keep them so that we can search for node globally
then in ossia.fuzzysearch, i'll add a root attribute to set the starting point of the search

Reply:
now we are listing all devices : local devices (ø.device), local client (ø.client), global max device and also the devices available through ZeroConf

now I'll add some mechanism to restrict the search to a given device, or a given list of nodes